File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.cpp
class Solution {
public:
    int findMin(vector<int> &nums) {
        int left = 0; 
        int right = nums.size() - 1; 

        while (left < right) { 
            int middle = (left + right) / 2; 

            if (nums[middle] > nums.back()) { 
                left = middle + 1; 
            }
            else { // nums[middle] <= nums[end] 
                right = middle; 
            }
        }

        return nums[left]; 
        
    }
};

File: Data Structures & Algorithms/longest-palindromic-substring/submission-1.py
from collections import heapq
class Solution:
    def longestPalindrome(self, s: str) -> str:
        # centers = len - 2 
        # start from middle if palindrome is found return 
        res = ""
        longest_word = 0 

        for center in range(len(s)): 
            # Odd centers 
            left, right = center, center 
            while left >= 0 and right < len(s) and s[left] == s[right]:
                left -= 1 
                right += 1
            
            if right - left > longest_word: 
                longest_word = right - left 
                res = s[left+1:right]

            # Even centers
            left, right = center, center + 1  
            while left >= 0 and right < len(s) and s[left] == s[right]:
                left -= 1 
                right += 1
            
            if right - left > longest_word: 
                longest_word = right - left 
                res = s[left+1:right]


        return res


            

        

        